### PR TITLE
feat: set InstalledFiles for DEB and RPM packages

### PR DESCRIPTION
**Pull request by lebauce**
## Description

This PR does 2 main things :
  - Currently, only apk package analyzer attaches the installed files to the `Package` structure. This PR does the same for DEB and RPM.
  - It puts this installed file collection behind an artifact option as not everyone may be interested in collecting these files.

## Related issues
- Close #XXX

## Related PRs
- [ ] #XXX
- [ ] #YYY

Remove this section if you don't have related PRs.

## Checklist
- [ ] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [ ] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by lebauce** [comment]
@knqyf263 No problem. The feature itself should work but I'm struggling with updating the fixtures of integration tests. Working on it

**Comment by lebauce** [comment]
@knqyf263 I fixed the tests and undrafted the PR

File: pkg/fanal/analyzer/analyzer.go [hunk 1]
```diff
@@ -146,6 +146,7 @@ type PostAnalysisInput struct {
 type AnalysisOptions struct {
 	Offline      bool
 	FileChecksum bool
+	CollectFiles bool
```
Comment (knqyf263) [inline pkg/fanal/analyzer/analyzer.go:149]: We can collect the files by default. I mean we may not need the option. What do you think?

Comment (lebauce) [reply pkg/fanal/analyzer/analyzer.go]: :+1: 
I updated the PR without the option

**Comment by knqyf263** [comment]
Thanks for updating the tests, but a massive list of expected values is not our intention. I created a PR to improve the RPM tests.
https://github.com/aquasecurity/trivy/pull/5567

I'll create a PR improving VM tests as well. After that, we can add better tests to this  PR.

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**